sky/shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -195,17 +200,23 @@ float getDevicePixelRatio() {
     public void destroy() {
         if (mPlatformServiceProviderBinding != null) {
             mPlatformServiceProviderBinding.unbind().close();
+            mPlatformServiceProviderProxy.close();
             mPlatformServiceProvider.unbindServices();
         }
 
         if (mViewServiceProviderBinding != null) {
             mViewServiceProviderBinding.unbind().close();
+            mViewServiceProviderProxy.close();
             mViewServiceProvider.unbindServices();
         }
 
         getHolder().removeCallback(mSurfaceCallback);
         nativeDetach(mNativePlatformView);
         mNativePlatformView = 0;
+
+        mSkyEngine.close();
+        mDartServiceProvider.close();
+        mFlutterAppMessages.close();
```

Comment:
These three LGTM, but I don't think we need to store or close the `mPlatformServiceProviderProxy` and `mViewServiceProviderProxy` objects.